web/client/plugins/LongitudinalProfileTool.jsx

```diff
@@ -0,0 +1,213 @@
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import GlobalSpinner from '../components/misc/spinners/GlobalSpinner/GlobalSpinner';
+import Main from './longitudinalProfile/Main';
+import UserMenuConnected from './longitudinalProfile/Menu';
+import { setControlProperty } from "../actions/controls";
+import {
+    addMarker,
+    changeCRS,
+    changeDistance,
+    changeGeometry,
+    changeReferential,
+    closeDock,
+    hideMarker,
+    showError,
+    setup,
+    tearDown,
+    toggleMaximize,
+    toggleMode
+} from "../actions/longitudinalProfile";
+import { warning } from '../actions/notifications';
+import {exportCSV} from "../actions/widgets";
+import * as epics from '../epics/longitudinalProfile';
+import longitudinalProfile from '../reducers/longitudinalProfile';
+import {
+    getSelectedLayer as getSelectedLayerSelector
+} from '../selectors/layers';
+import {
+    currentLocaleSelector,
+    currentMessagesSelector
+} from '../selectors/locale';
+import {
+    chartTitleSelector,
+    crsSelectedDXFSelector,
+    configSelector,
+    dataSourceModeSelector,
+    distanceSelector,
+    infosSelector,
+    isDockOpenSelector,
+    isInitializedSelector,
+    isLoadingSelector,
+    isMaximizedSelector,
+    isParametersOpenSelector,
+    isSupportedLayerSelector,
+    pointsSelector,
+    projectionSelector,
+    referentialSelector
+} from '../selectors/longitudinalProfile';
+import {
+    dockStyleSelector,
+    helpStyleSelector,
+    boundingSidebarRectSelector
+} from '../selectors/maplayout';
+
+import { createPlugin } from '../utils/PluginsUtils';
+
+/**
+ * Plugin for generating a chart with longitudinal profile
+ * @name LongitudinalProfileTool
+ * @memberof plugins
+ * @class
+ * @prop {Object} cfg.config the plugin configuration
+ * @prop {string} cfg.config.chartTitle the default title of the chart
+ * @prop {number} cfg.config.defaultDistance the default distance value in meters
+ * @prop {string} cfg.config.identifier the profile to use in the wps request, defaulted to gs:LongitudinalProfile
+ * @prop {string} cfg.config.defaultReferentialName the default referential name
+ * @prop {Object[]} cfg.config.referentials the layers that can be used as referentials
+ * @prop {string[]} cfg.filterAllowedCRS the allowed crs to be proposed when dropping a DXF file, (needs to be supported by mapstore)
+ * @prop {Object} cfg.additionalCRS the crs object that allow to define also a label
+ *
+ * @example
+ *
+ * {
+ *   "name": "LongitudinalProfileTool",
+ *   "cfg": {
+ *     "config": {
+ *       "chartTitle": "Longitudinal profile",
+ *       "defaultDistance": 75,
+ *       "defaultReferentialName": "sfdem",
+ *       "referentials": [{
+ *          "layerName": "sfdem",
+ *          "title": "sfdem"
+ *        }]
+ *      },
+ *      "filterAllowedCRS": ["EPSG:4326", "EPSG:3857"],
+ *      "additionalCRS": {
+ *        "EPSG:3003": { "label": "EPSG:3003" }
+ *       },
```

Comment:
Just a note also related to the crs/mouse configuration. This use of double config is a bit confusing and it seems it was introduces to ensure that also the default crs were included. I think that using a single properties like availableCRS that list all the crs a user want to configure is much clear:
```
"avaliableCRS": [
  { "label": "EPSG:4326", "value": "EPSG:4326" },
  { "label": "EPSG:3857", "value": "EPSG:3857" },
  { "label": "EPSG:3003", "value": "EPSG:3003" }
]
```
Anyway I think that crs configuration in general should be centralize if there is not a specific need for a plugin. An example all the configured projectDefs + the default EPSG:4326 and EPSG:3857 available for all the plugins that need them.

Reply:
I totally agree with you @allyoucanmap. I've already discussed also with @MV88 about that but this behavior is the one used currently also for other plugins and there isn't the opportunity to change this as part of this work. A new issue can be created anyway.

Reply:
i have created this issue

https://github.com/geosolutions-it/MapStore2/issues/9249

please @tdipisa review the milestone assigned